MusicCrawler.Lib/Environment/SpotifyApi.cs

```diff
@@ -0,0 +1,121 @@
+ *      https://developer.spotify.com/documentation/web-api/tutorials/code-flow
+ *      https://developer.spotify.com/documentation/web-api/tutorials/client-credentials-flow
+ */
+public class SpotifyApi
+{
+    private readonly HttpClient _httpClient;
+
+    private readonly string _baseUri;
+
+    // The client ID from https://developer.spotify.com/dashboard
+    private const string client_id = "267c94026025449b8013ddde6d959e13";
+
+    // The client secret from https://developer.spotify.com/dashboard
+    private const string client_secret = "92c88db9315545e38989ca8cc4cad2ad";
+
+    // This is the page that the user will be sent to after clicking "accept" from the html given by accounts.spotify.com/authorize.
+    private const string redirect_uri = "http://localhost/";
+
+    public SpotifyApi()
+    {
+        this._baseUri = "https://api.spotify.com";
+        this._httpClient = new HttpClient();
+    }
+
+    public SpotifyApi(string baseUri)
+    {
+        this._baseUri = baseUri;
+        this._httpClient = new HttpClient();
+    }
+
+
+    /**
+     * responds with an html for a page where the user can click "accept".
+     * Might be useful if we want to access information specific to a spotify user.
+     */
+    public async Task<string> OAuthLogin()
+    {
+        var state = RandomStringGenerator.GenerateRandomString(16);
+        var scope = "user-read-private user-read-email";
+
+        var queryParams = HttpUtility.ParseQueryString(string.Empty);
+        queryParams["response_type"] = "code";
+        queryParams["client_id"] = client_id;
+        queryParams["scope"] = scope;
+        queryParams["redirect_uri"] = redirect_uri;
+        queryParams["state"] = state;
+
+        var url = $"https://accounts.spotify.com/authorize?{queryParams}";
+
+        Console.WriteLine($"url:{url}");
+
+        var response = await _httpClient.GetStringAsync(url);
+
+        return response;
+    }
+
+    /**
+     * retrieves an oAuth token without user input and therefore has no user-related permissions.
+     */
+    public async Task<string> NonUserOAuthToken()
+    {
+        var authHeader = Convert.ToBase64String(Encoding.UTF8.GetBytes($"{client_id}:{client_secret}"));
+
+        var requestData = new FormUrlEncodedContent(new[]
+        {
+            new KeyValuePair<string, string>("grant_type", "client_credentials")
+        });
+
+        var request = new HttpRequestMessage
+        {
+            Method = HttpMethod.Post,
+            RequestUri = new Uri("https://accounts.spotify.com/api/token"),
+            Headers = { { "Authorization", $"Basic {authHeader}" } },
+            Content = requestData
+        };
+
+        var response = await _httpClient.SendAsync(request);
+        var responseBody = await response.Content.ReadAsStringAsync();
+        var token = JsonConvert.DeserializeObject<AccessTokenResponse>(responseBody)?.access_token;
+
+        Debug.Assert(token != null, nameof(token) + " != null");
```

Comment:
I havent used many Debug.Asserts in my code (maybe I should?).   But I do think it eventually gets removed during a release build.   Usually in these situations I'm more familiar with seeing something like
`return token ?? throw new NullReferenceException();`